Documentation/contributing.rst

```diff
@@ -1482,47 +1497,90 @@ GitHub issues using the process below:
+   CI-Bug, Flake, or a Regression as defined in the table above.
+
+   #. Search `GitHub issues <https://github.com/cilium/cilium/issues?utf8=%E2%9C%93&q=is%3Aissue+>`_
+      to see if bug is already filed. Make sure to also include closed issues in
+      your search as a CI issue can be considered solved and then re-appears.
+      Good search terms are:
+
+      - The test name, e.g.
+      ::
+
+          k8s-1.7.K8sValidatedKafkaPolicyTest Kafka Policy Tests KafkaPolicies (from (k8s-1.7.xml))
+
+      - The line on which the test failed, e.g.
+      ::
+
+          github.com/cilium/cilium/test/k8sT/KafkaPolicies.go:202
+
+      - The error message, e.g.
+      ::
+
+          Failed to produce from empire-hq on topic deathstar-plan
+
+#. If a corresponding GitHub issue exists, update it with:
+
+   #. A link to the failing Jenkins build (note that the build information is
+      eventually deleted).
+   #. Attach the zipfile downloaded from Jenkins with logs from the failing
+      tests. A zipfile for all tests is also available.
+   #. Check how much time has passed since the last reported occurrence of this
+      failure and move this issue to the correct column in the `CI flakes
+      project <https://github.com/cilium/cilium/projects/8>`_ board.
+
+#. If no existing GitHub issue was found, file a `new GitHub issue <https://github.com/cilium/cilium/issues/new>`_:
 
    #. Attach zipfile downloaded from Jenkins with logs from failing test
-   #. If failure is a real bug:
+   #. If the failure is a new regression or a real bug:
 
       #. Title: ``<Short bug description>``
       #. Labels ``kind/bug`` and ``needs/triage``.
-   #. If failure is a CI test flake or if unsure:
+
+   #. If failure is a new CI-Bug, Flake or if you are unsure:
 
       #. Title ``CI: <testname>: <cause>``, e.g. ``CI: K8sValidatedPolicyTest Namespaces: cannot curl service``
       #. Labels ``kind/bug/CI`` and ``needs/triage``
+      #. Include a link to the failing Jenkins build (note that the build information is
+         eventually deleted).
+      #. Attach zipfile downloaded from Jenkins with logs from failing test
+      #. Include the test name and whole Stacktrace section to help others find this issue.
       #. Add issue to `CI flakes project <https://github.com/cilium/cilium/projects/8>`_
 
-#. Edit the description of Jenkins build to mark it as triaged
+   .. note::
 
-   #. Login -> Click on build -> Edit Build Information
-   #. Use the table below as a template
+      Be extra careful when you see a new flake on a PR, and want to open an
+      issue. It's much more difficult to debug these without context around the
+      PR and the changes it introduced. When creating an issue for a PR flake,
+      include a description of the code change, the PR, or the diff. If it
+      isn't related to the PR, then it should already happen in master, and a
+      new issue isn't needed. 
```

Comment:
Awesome!

Reply:
Here it'll be great if the top information of the build can be added, like: 

```
GitHub pull request #4881 of commit b3f939b7c3d27033142f8983fb6a8983b949d4c8, no merge conflicts.
GitHub pull request #4881 of commit b3f939b7c3d27033142f8983fb6a8983b949d4c8, no merge conflicts.
Setting status of b3f939b7c3d27033142f8983fb6a8983b949d4c8 to PENDING with url https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5074/ and message: 'Build started for merge commit.'
Using context: Cilium-Ginkgo-Tests
Checking out git http://github.com/cilium/cilium into /var/jenkins_home/workspace/Cilium-PR-Ginkgo-Tests-Validated@script to read ginkgo.Jenkinsfile
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url http://github.com/cilium/cilium # timeout=10
Fetching upstream changes from http://github.com/cilium/cilium
 > git --version # timeout=10
using GIT_ASKPASS to set credentials 
 > git fetch --tags --progress http://github.com/cilium/cilium +refs/pull/*:refs/remotes/origin/pr/*
 > git rev-parse refs/remotes/origin/pr/4881/merge^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/pr/4881/merge^{commit} # timeout=10
JENKINS-19022: warning: possible memory leak due to Git plugin usage; see: https://wiki.jenkins-ci.org/display/JENKINS/Remove+Git+Plugin+BuildsByBranch+BuildData
Checking out Revision 1ea5fe2d4d8a401ef6e5678c5c307b42bf086289 (refs/remotes/origin/pr/4881/merge)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 1ea5fe2d4d8a401ef6e5678c5c307b42bf086289
Commit message: "Merge b3f939b7c3d27033142f8983fb6a8983b949d4c8 into 92243feb5df4956e87874b94ba77278d67ba3206"
First time build. Skipping changelog.
Running in Durability level: MAX_SURVIVABILITY
Loading library cilium@master
Attempting to resolve master from remote references...
 > git --version # timeout=10
 > git ls-remote -h https://github.com/cilium/Jenkins-library.git # timeout=10
Found match: refs/heads/master revision ce671cad8196752c0b46844fa59d39f048780bb2
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/cilium/Jenkins-library.git # timeout=10
Fetching without tags
Fetching upstream changes from https://github.com/cilium/Jenkins-library.git
 > git --version # timeout=10
 > git fetch --no-tags --progress https://github.com/cilium/Jenkins-library.git +refs/heads/*:refs/remotes/origin/*
Checking out Revision ce671cad8196752c0b46844fa59d39f048780bb2 (master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f ce671cad8196752c0b46844fa59d39f048780bb2
Commit message: "Added BuildAsync and BuildIfMatch"
 > git rev-list --no-walk ce671cad8196752c0b46844fa59d39f048780bb2 # timeout=10
[Pipeline] node
Still waiting to schedule task
Waiting for next available executor on baremetal
Running on jenkins-node-0 in /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Declarative: Checkout SCM)
[Pipeline] checkout
Cloning the remote Git repository
Cloning repository http://github.com/cilium/cilium
 > git init /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated # timeout=10
Fetching upstream changes from http://github.com/cilium/cilium
 > git --version # timeout=10
using GIT_ASKPASS to set credentials 
 > git fetch --tags --progress http://github.com/cilium/cilium +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin.url http://github.com/cilium/cilium # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url http://github.com/cilium/cilium # timeout=10
Fetching upstream changes from http://github.com/cilium/cilium
using GIT_ASKPASS to set credentials 
 > git fetch --tags --progress http://github.com/cilium/cilium +refs/pull/*:refs/remotes/origin/pr/*
 > git rev-parse refs/remotes/origin/pr/4881/merge^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/pr/4881/merge^{commit} # timeout=10
JENKINS-19022: warning: possible memory leak due to Git plugin usage; see: https://wiki.jenkins-ci.org/display/JENKINS/Remove+Git+Plugin+BuildsByBranch+BuildData
Checking out Revision 1ea5fe2d4d8a401ef6e5678c5c307b42bf086289 (refs/remotes/origin/pr/4881/merge)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 1ea5fe2d4d8a401ef6e5678c5c307b42bf086289
Commit message: "Merge b3f939b7c3d27033142f8983fb6a8983b949d4c8 into 92243feb5df4956e87874b94ba77278d67ba3206"
```

Maybe I can check if any plugin that does something like this.

Reply:
@eloycoto  I can add manual instructions, but I'm not sure how consistent it is. Would this be correct?
Paste in the start of the build logs, between
`GitHub pull request #4881 of commit b3f939b7c3d27033142f8983fb6a8983b949d4c8, no merge conflicts.`
and 
`Commit message: "Merge b3f939b7c3d27033142f8983fb6a8983b949d4c8 into 92243feb5df4956e87874b94ba77278d67ba3206"`

Wouldn't this information already be in the file bundle we create? I guess it's nicer to see it directly in the issue?